include/avif/avif.h

```diff
@@ -862,6 +862,10 @@ AVIF_API avifResult avifImageAllocatePlanes(avifImage * image, avifPlanesFlags p
 AVIF_API void avifImageFreePlanes(avifImage * image, avifPlanesFlags planes);           // Ignores already-freed planes
 AVIF_API void avifImageStealPlanes(avifImage * dstImage, avifImage * srcImage, avifPlanesFlags planes);
 
+// Add arbitrary (opaque) properties to the image
+AVIF_API avifResult avifImageAddOpaqueProperty(avifImage * image, const uint8_t boxtype[4], const uint8_t * data, size_t dataSize);
+AVIF_API avifResult avifImageAddUUIDProperty(avifImage * image, const uint8_t uuid[16], const uint8_t * data, size_t dataSize);
```

Comment:
Did you consider using only one signature leveraging the existing `avifImageItemProperty` definition?

```c
AVIF_API avifResult avifImageAddProperty(avifImage * image, const avifImageItemProperty * property);
```

The downside is the need for `avifImageItemPropertyCreate()` and `avifImageItemPropertyDestroy()` so it may be worse.

Alternatively it could reuse the struct members but not the struct itself:

```c
// Appends an avifImageItemProperty to the image properties.
// The arguments have the same semantics as the fields of avifImageItemProperty.
AVIF_API avifResult avifImageAddProperty(avifImage * image, const uint8_t boxtype[4], const uint8_t usertype[16], avifROData boxPayload);
```

Reply:
I did think about that, but it seemed somewhat cleaner to have two different methods. It just makes it harder to use incorrectly.